modules/correlation/groupingby.c

```diff
@@ -30,7 +32,8 @@ typedef struct _GroupingBy
   FilterExprNode *where_condition_expr;
   FilterExprNode *having_condition_expr;
   gchar *prefix;
-  gint clone_id;
+  guint clone_id;
```

Comment:
`clone_id` is initialized to 0 because of `g_new0()`. Fortunately a newly created `IdCounter` returns 1 for the first `get_next_id()` call. This works, but the knowledge is in two different parts of the code. I would suggest to always set `clone_id` with the help of `id_counter`. This needs `id_counter` to be allocated in `new()` (there is another argument in my other comment to move it there) and we need to make sure that `IdCounter` returns 0 for the `get_next_id()` (initing its `id` field to -1 would suffice).

Reply:
I think a newly created `IdCounter` will return 0 on the first call to `get_next_id()` because the private member `id` is initialized to 0 and and the `return self->id++` line in `get_next_id()` will only increment the value after returning it.

However, I have made the changes to allocate `IdCounter` in `new()` as you have suggested.